src/core/network/nfts.ts

```diff
@@ -0,0 +1,118 @@
+  SimpleHashNFT,
+} from '../types/nfts';
+
+import { RainbowFetchClient, rainbowFetch } from './internal/rainbowFetch';
+
+interface SimpleHashCollectionsResponse {
+  collections: SimpleHashCollectionDetails[];
+  next_cursor: string | null;
+}
+
+interface SimpleHashNftsResponse {
+  nfts: SimpleHashNFT[];
+  next_cursor: string | null;
+}
+
+const nftApi = new RainbowFetchClient({
+  baseUrl: `https://${process.env.NFT_API_URL}/api/v0`,
+  headers: {
+    'x-api-key': process.env.NFT_API_KEY || '',
+  },
+});
+
+export const fetchNftCollections = async ({
+  address,
+  chains,
+}: {
+  address: string;
+  chains: ChainName[];
+}) => {
+  try {
+    let cursor: string | null = 'none';
+    let collections: SimpleHashCollectionDetails[] = [];
+    while (cursor) {
+      const response: {
+        data: SimpleHashCollectionsResponse;
+        headers: Headers;
+        status: number;
+      } =
+        // eslint-disable-next-line no-await-in-loop
+        await nftApi.get('/nfts/collections_by_wallets_v2', {
+          params: {
+            chains: chains.join(','),
+            ...(cursor && cursor !== 'none' ? { cursor } : {}),
+            wallet_addresses: address,
+          },
+        });
+      collections = [...collections, ...(response.data?.collections || [])];
+      cursor = response.data?.next_cursor;
+    }
+    return collections;
+  } catch (e) {
+    logger.error(new RainbowError('Fetch NFT Collections: '), {
+      message: (e as Error)?.message,
+    });
+    return [];
+  }
+};
+
+export const fetchNfts = async ({
+  address,
+  chains,
+  collectionIds,
+}: {
+  address: string;
+  chains: ChainName[];
+  collectionIds: string[];
+}) => {
+  try {
+    let cursor: string | null = 'none';
+    let nfts: SimpleHashNFT[] = [];
+    while (cursor) {
+      const response: {
+        data: SimpleHashNftsResponse;
+        headers: Headers;
+        status: number;
+      } =
+        // eslint-disable-next-line no-await-in-loop
+        await nftApi.get('/nfts/owners', {
+          params: {
+            chains: chains.join(','),
+            collection_ids: collectionIds.join(','),
+            ...(cursor && cursor !== 'none' ? { cursor } : {}),
+            wallet_addresses: address,
+          },
+        });
+      nfts = [...nfts, ...(response.data?.nfts || [])];
+      cursor = response.data?.next_cursor;
+    }
+    return nfts;
+  } catch (e) {
+    logger.error(new RainbowError('Fetch NFTs: '), {
+      message: (e as Error)?.message,
+    });
+    return [];
+  }
+};
+
+export const fetchPolygonAllowList =
+  async (): Promise<PolygonAllowListDictionary> => {
+    const allowList = await rainbowFetch<{ data: { addresses: string[] } }>(
+      'https://metadata.p.rainbow.me/token-list/137-allowlist.json',
+      { method: 'get' },
+    );
+    const polygonAllowListDictionary = allowList.data?.data?.addresses?.reduce(
+      (allowListDict, tokenAddress) => {
+        allowListDict[tokenAddress] = true;
```

Comment:
just to make sure the api return `tokenAddress` all lowercase and not checksummed, as that would break the filter later

Reply:
are we good here?